src/main.rs

```diff
@@ -114,97 +123,102 @@ fn post_judgement(
     form: Form<AudioSubm>,
     audio_mx: State<AudioMX>,
 ) -> Result<Flash<Redirect>, Flash<Redirect>> {
-    let mut map = audio_mx.mp.lock().expect("locking");
-    let lister = match map.get_mut(&subdir) {
-        None => {
-            return Err(Flash::error(
-                Redirect::to(uri![categorize_genre: subdir]),
-                "Subdir not registered",
-            ))
-        }
-        Some(x) => x,
-    };
+    let mut map = audio_mx.lock().unwrap();
+
+    let lister = map.get_mut(&subdir).ok_or_else(|| {
+        Flash::error(
+            Redirect::to(uri![categorize_genre: &subdir]),
+            "Subdir not registered",
+        )
+    })?;
+
     let dest_dir = if accept { &subdir } else { "reject" };
-    match lister.move_file_and_remove(
-        &form.sec,
-        PathBuf::from(GLOBAL_BASE)
-            .join(&subdir)
-            .join(&form.filename),
-        PathBuf::from(GLOBAL_DEST)
-            .join(&dest_dir)
-            .join(&form.filename),
-    ) {
-        Err(e) => {
+
+    lister
+        .move_file_and_remove(
+            &form.sec,
+            PathBuf::from(GLOBAL_BASE)
+                .join(&subdir)
+                .join(&form.filename),
+            PathBuf::from(GLOBAL_DEST)
+                .join(&dest_dir)
+                .join(&form.filename),
+        )
+        .map(|_| Flash::success(Redirect::to(uri![categorize_genre: &subdir]), "success!"))
+        .map_err(|e| {
             println!("{}", e.to_string());
-            return Err(Flash::error(
-                Redirect::to(uri![categorize_genre: subdir]),
-                e.to_string(),
-            ));
-        }
-        Ok(_) => (),
-    }
-    Ok(Flash::success(
-        Redirect::to(uri![categorize_genre: subdir]),
-        "success!",
-    ))
+            Flash::error(Redirect::to(uri![categorize_genre: &subdir]), e.to_string())
+        })
 }
 
 #[get("/file/<subdir>/<name..>")]
 fn get_file(subdir: String, name: PathBuf) -> Option<NamedFile> {
     NamedFile::open(Path::new(&GLOBAL_BASE).join(subdir).join(name)).ok()
 }
 
-fn setup(subdirs: Vec<String>, timeout_mins: u64) -> io::Result<rocket::Rocket> {
-    let mp = init_map(
-        PathBuf::from(&GLOBAL_BASE),
-        subdirs,
-        Duration::from_secs(timeout_mins * 60),
-    )?;
+fn setup(subdirs: Vec<String>, timeout: Duration) -> io::Result<rocket::Rocket> {
```

Comment:
With `timeout_mins: u64`, the call-site looks weird: `setup(dirs, 5)` -- exactly the issue the `Duration::from_X` naming scheme is designed to avoid. But I understand wanting to deal with minutes, which is why I added the `DurationExt` helper trait above; then, we can call `setup(dirs, Duration::from_mins(5))`, which is much clearer.